src/List/ItemList.js

```diff
@@ -82,6 +86,187 @@ const Items = props => {
     }
   };
 
+  //comparison method for sorting by purchase date.
+  const comparison = (itemA, itemB) => {
+    let comparison = 0;
+    if (itemA.nextPurchaseDate > itemB.nextPurchaseDate) {
+      comparison = 1;
+    } else if (itemA.nextPurchaseDate < itemB.nextPurchaseDate) {
+      comparison = -1;
+    }
+    return comparison;
+  };
+
+  //update the item.nextPurchaseDate to whole integer
+
+  const buySoon = (
+    <div>
+      <h1 className={classes.buySoon}> Buy Soon </h1>
+      <ul>
+        {dbItems
+          .filter(item => {
+            if (item.name !== undefined) {
+              return item.name
+                .toLowerCase()
+                .includes(filterInput.toLowerCase());
+            }
+          })
+          .sort(comparison)
+          .map((item, index) => {
+            if ((item.nextPurchaseDate || item.frequency) < 7) {
+              return (
+                <li key={index}>
+                  <label>
+                    <input
+                      type="checkbox"
+                      name="isPurchased"
+                      checked={is24Hours(item)}
+                      onChange={e => handleMarkPurchased(e, item)}
+                      value={item.name}
+                    />
+                    {item.name}
+                    <button
+                      role="button"
+                      ariaLabel="delete item"
+                      onClick={e => handleDelete(e)}
+                      value={item.name}
+                    >
+                      Delete Item{" "}
+                    </button>
+                  </label>
+                </li>
+              );
+            }
+          })}
+      </ul>
+    </div>
+  );
+
+  const buyKindaSoon = (
+    <div>
+      <h1 className={classes.buyKindaSoon}> Buy Kinda Soon </h1>
+      <ul>
+        {dbItems
+          .filter(item => {
+            if (item.name !== undefined) {
+              return item.name
+                .toLowerCase()
+                .includes(filterInput.toLowerCase());
+            }
+          })
+          .sort(comparison)
+          .map((item, index) => {
+            if (
+              (item.nextPurchaseDate || item.frequency) >= 7 &&
+              (item.nextPurchaseDate || item.frequency) < 30
+            ) {
+              return (
+                <li key={index}>
```

Comment:
My thought with refactoring is that everything in this `<li>` seems to be repeated, so you can extract this into an external function that only returns JSX and then return it in here (passing in the relevant arguments)

Someone else might have mentioned this already, didn't read the other reviews yet.

Reply:
> an external function that only returns JSX and then return it in here

I want to point out that a React component is a function, so I would highly recommend you create a component to articulate this list item to stay idiomatic to React:

```jsx
function ShoppingListItem(props) {
  const { item, handleDelete, handleMarkPurchased } = props;

  return (
    <li>
      <label>
        <input
          type="checkbox"
          name="isPurchased"
          checked={is24Hours(item)}
          onChange={e => handleMarkPurchased(e, item)}
          value={item.name}
        />
        {item.name}
        <button onClick={e => handleDelete(e)} value={item.name}>
          X{" "}
        </button>
      </label>
    </li>     
  )
}
```